lib/pa_ess/utilities.ex

```diff
@@ -545,8 +545,15 @@ defmodule PaEss.Utilities do
 
   @spec prediction_approaching?(Predictions.Prediction.t(), boolean()) :: boolean()
   def prediction_approaching?(prediction, terminal?) do
+    secs_to_announce = secs_to_announce_approaching(prediction.route_id)
+
     !terminal? and !prediction.stopped_at_predicted_stop? and
-      prediction_seconds(prediction, terminal?) in 0..45
+      prediction_seconds(prediction, terminal?) in 0..secs_to_announce
+  end
+
+  @spec secs_to_announce_approaching(String.t()) :: integer()
+  defp secs_to_announce_approaching(route_id) do
+    if String.starts_with?(route_id, "Green"), do: 30, else: 45
```

Comment:
Picking your brain about this: when does it make sense to do something like this versus pattern match on `route_id` on something like `Green- <> _`?

Reply:
Apologies, that is what I meant. 

Rereading my comment, I swear I thought I had something about matching withing the function, but I think Github did something weird and reverted it to a previous comment. 

But yep, what Cora said.

Reply:
gotcha thanks for pointing out! This is exactly the type of way I would've liked to implement, and I'm updating it to this pattern matching now